Composer/packages/client/src/CreationFlow/DefineConversation/index.js

```diff
@@ -59,6 +75,22 @@ export function DefineConversation(props) {
     });
   };
 
+  function getDefaultName() {
+    let i = 0;
+    const bot = templateId || botName;
+    let defaultName = `${bot}-${i}`;
+
+    while (
+      bots.findIndex(bot => {
```

Comment:
Add a maximum retry for defensive programming reason. I.e. if index > 10000, we shouldn't proceed and throw an error. Otherwise, we will hang the app.